awx/ui_next/src/components/Search/Search.jsx

```diff
@@ -65,78 +77,107 @@ class Search extends React.Component {
     this.handleSearch = this.handleSearch.bind(this);
   }
 
-  handleDropdownToggle(isSearchDropdownOpen) {
+  handleDropdownToggle (isSearchDropdownOpen) {
     this.setState({ isSearchDropdownOpen });
   }
 
-  handleDropdownSelect({ target }) {
+  handleDropdownSelect ({ target }) {
     const { columns } = this.props;
     const { innerText } = target;
 
     const { key: searchKey } = columns.find(({ name }) => name === innerText);
     this.setState({ isSearchDropdownOpen: false, searchKey });
   }
 
-  handleSearch() {
-    const { searchValue } = this.state;
+  handleSearch (e) {
+    // keeps page from fully reloading
+    e.preventDefault();
+
+    const { searchKey, searchValue } = this.state;
     const { onSearch } = this.props;
 
-    onSearch(searchValue);
+    // TODO: probably not _always_ add icontains.  I don't think icontains works for numbers.
```

Comment:
:+1: this seems like a good TODO to handle in a follow-up pr. If i'm following the problem correctly, this is where we start to get into how query params might need to vary depending on _which_ field we're searching / filtering on.

I'm wondering if the `types.js` definitions @keithjgrant has begun to setup in `src/.types.jsx` is something we can leverage here.